chipsec/hal/psp.py

```diff
@@ -0,0 +1,119 @@
+# CHIPSEC: Platform Security Assessment Framework
+# Copyright (c) 2024, AMD Corporation
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; Version 2.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# Contact information:
+# chipsec@amd.com
+#
+
+"""
+"""
+
+import struct
+from collections import namedtuple
+import itertools
+import time
+from typing import List, Tuple, Optional
+from chipsec.library.logger import logger, pretty_print_hex_buffer
+from chipsec.library.file import write_file
+from chipsec.library.exceptions import OsHelperError
+from chipsec.library.defines import is_all_ones, MASK_16b, MASK_32b, MASK_64b, BOUNDARY_4KB
+
+class PSP:
+    SMU_INDEX_ADDR = 0xb8
```

Comment:
It's really not nice to go around the kernels back with these mailboxes.

There is a debugfs interface on the mailing list that has proper locking protection.

https://lore.kernel.org/all/20241023172150.659002-17-yazen.ghannam@amd.com/

Reply:
This is HAL code, which will also be used on Windows and UEFI. It's hardly the only part where Chipsec modules do unreliable accesses that could race with the kernel I believe (eg. I/O ports) so I don't think it should be blocker.

Reply:
I see.  Well FYI this type of access will taint the kernel in the future, and eventually want to block it to force moving to the debugfs interface.

Reply:
And BTW this naming is totally wrong, this isn't a PSP or SMU address.  This is an SMN router that you're interacting with.